bots/inspect-queue

```diff
@@ -21,39 +21,50 @@ MAX_PRIORITY = 9
 
 import argparse
 import sys
-import time
+import ssl
 
-import amqp
+import pika
 
 def main():
     parser = argparse.ArgumentParser(description='Read and print messages from the queue without acknowleding them')
-    parser.add_argument('--amqp', default='localhost',
-                        help='The URL of the AMQP server to consume from')
-    parser.add_argument('--queue', default='tasks',
-                        help='The name of the queue to consume from')
-    parser.add_argument('-n', '--number', default=0,
-                        help='Number of queue items to show, starting from the front (0=all)')
-    parser.add_argument('-t', '--timeout', default=0.5,
-                        help='Time to wait for messages')
+    parser.add_argument('--amqp', default='localhost:5671',
+                        help='The URL of the AMQP server to consume from (default: %(default)s)')
     opts = parser.parse_args()
 
-    connection = amqp.Connection(host=opts.amqp)
-    connection.connect()
-    channel = connection.channel()
     try:
-        channel.queue_declare(queue=opts.queue, passive=True, auto_delete=False)
-    except amqp.exceptions.NotFound:
-        sys.stdout.write('queue is empty\n')
-        return 0
-    channel.basic_qos(0, opts.number, True)
-
-    def callback(msg):
-        sys.stdout.write("{0}\n".format(msg.body))
-        msg.channel.basic_reject(msg.delivery_tag, requeue=True)
-
-    channel.basic_consume(callback=callback, queue=opts.queue)
-    time.sleep(opts.timeout)
-    sys.stdout.flush()
+        host, port = opts.amqp.split(':')
+    except ValueError:
+        sys.stderr.write('Please format URL as host:port\n')
+        return 1
```

Comment:
It's the simplest thing, but `parser.error('...') is a bit more idiomatic, and also exits the program. And it's really not an URL :-)